src/Features/Core/Portable/Diagnostics/IDiagnosticAnalyzerService.cs

```diff
@@ -55,18 +55,18 @@ internal interface IDiagnosticAnalyzerService
 
         /// <summary>
         /// try to return up to date diagnostics for the given span for the document.
-        /// 
+        ///
         /// it will return true if it was able to return all up-to-date diagnostics.
         ///  otherwise, false indicating there are some missing diagnostics in the diagnostic list
         /// </summary>
         Task<bool> TryAppendDiagnosticsForSpanAsync(Document document, TextSpan range, List<DiagnosticData> diagnostics, bool includeSuppressedDiagnostics = false, CancellationToken cancellationToken = default);
 
         /// <summary>
         /// return up to date diagnostics for the given span for the document
-        /// 
+        ///
         /// this can be expensive since it is force analyzing diagnostics if it doesn't have up-to-date one yet.
         /// </summary>
-        Task<IEnumerable<DiagnosticData>> GetDiagnosticsForSpanAsync(Document document, TextSpan range, bool includeSuppressedDiagnostics = false, CancellationToken cancellationToken = default);
+        Task<IEnumerable<DiagnosticData>> GetDiagnosticsForSpanAsync(Document document, TextSpan range, bool includeSuppressedDiagnostics = false, string diagnosticId = null, CancellationToken cancellationToken = default);
```

Comment:
>diagnosticId [](start = 154, length = 12)

Rename to `diagnosticIdOpt` and also add comment explaining the semantics for null and non-null value.

Reply:
fixed